			     BASH PATCH REPORT
			     =================

Bash-Release:	5.2
Patch-ID:	bash52-016

Bug-Reported-by:	F G <frank.graziano@gmail.com>
Bug-Reference-ID:	<CAOhYt35M5VctK+xAPu=Gy_UzzGmHedWPJE4q+kL4UHF_6Nb1kA@mail.gmail.com>
Bug-Reference-URL:

Bug-Description:

If an expression in an arithmetic for loop expands to NULL, the shell
would crash.

Patch (apply with `patch -p0'):

*** ../bash-5.2-patched/execute_cmd.c	Thu Feb 23 14:15:05 2023
--- execute_cmd.c	Mon Feb 27 17:53:08 2023
***************
*** 3051,3055 ****
    if (l->next)
      free (expr);
!   new = make_word_list (make_word (temp), (WORD_LIST *)NULL);
    free (temp);
  
--- 3051,3055 ----
    if (l->next)
      free (expr);
!   new = make_word_list (make_word (temp ? temp : ""), (WORD_LIST *)NULL);
    free (temp);
  
*** ../bash-5.2/patchlevel.h	2020-06-22 14:51:03.000000000 -0400
--- patchlevel.h	2020-10-01 11:01:28.000000000 -0400
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 15
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 16
  
  #endif /* _PATCHLEVEL_H_ */
